server/models/index.js

```diff
@@ -8,9 +8,34 @@ module.exports = {
         callback(err, results);
       });
     },
-    addNew(params, callback = () => {}) {
+    addNew(productId, params, callback = () => {}) {
       const sql = 'INSERT INTO products (productId, name, price, prime, imageUrl, numReviews, avgRating) VALUES (?, ?, ?, ?, ?, ?, ?)';
-      db.query(sql, params, (err, results) => {
+      // console.log(params);
+      db.query(sql, [productId, ...params], (err, results) => {
+        callback(err, results);
+      });
+    },
+    getProduct(productId, callback) {
+      const sql = 'SELECT * FROM products WHERE productId = ?';
+      db.query(sql, productId, (err, results) => {
+        callback(err, results);
+      });
+    },
+    async updateProduct(productId, newProduct, callback) {
+      const sql1 = 'DELETE FROM productcategories WHERE id_products = (SELECT id FROM products WHERE productId = ?)';
+      await db.query(sql1, [productId], (err, results) => { // TODO: return both results from the final query
+        if (err) {
+          callback(err);
+        }
+      });
+      const sql2 = 'REPLACE INTO products (productId, name, price, prime, imageUrl, numReviews, avgRating) VALUES (?, ?, ?, ?, ?, ?, ?)';
```

Comment:
Consider breaking this off right before VALUES so that the line doesn't overflow.